wrangler-core/src/main/java/io/cdap/wrangler/executor/RecipePipelineExecutor.java

```diff
@@ -161,4 +186,38 @@ private List<Directive> getDirectives() throws RecipeException {
     }
     return directives;
   }
+
+  private Schema generateOutputSchema(Schema inputSchema, List<Row> output) throws RecordConvertorException {
+    Set<Field> outputFields = new HashSet<>();
```

Comment:
I don't think Field Implements hashcode/equals. Also this way you loose the order

Reply:
Fixed by using a LinkedHashMap to map potentially multiple schemas to one field name (the last one is taken) and also preserve insertion order (current behaviour when schema is generated)